1030: guard: To fix the record ID to contain the prefix '0x'

deepakala-k opened the pull request:
guard -l command : Formatted the record ID to display 0x and the header space to align

test:

root@rain104bmc:~# /tmp/guard -l
ID         | ERROR    | Type     | Path
0x10000000 | 00000000 | manual   | physical:sys-0/node-0/proc-1

Signed-off-by: deepakala-k <deepakala.karthikeyan@ibm.com>
Change-Id: I9c67945dea8a3098285e7270c67fe5697b25cea2

deepakala-k committed:
guard: Appending 0x for the ID and Error field

- Currently, the id was displayed in the hex value but, the user
  does not know about the format while displaying it, and the user
  need to pass format value if wants to invalidate a record.

- Similarly, the error log id was displayed in the hex value without
  format, so the user needs to know about the format if the user
  wants to get error log details through another tool ("peltool").

- So, appending the "0x" as a prefix for those field values.

Tested:

- Without Fix:

```
guard -l
ID       | ERROR    |  Type  | Path
...
0000000c | 00000000 | manual | physical:sys-0/node-0/proc-1

guard -i 0000000c
Could not convert: --invalidate = 0000000c
```
 - With Fix:

```
guard -l
ID       | ERROR    |  Type  | Path
...
0x0000000c | 00000000 | manual | physical:sys-0/node-0/proc-1

guard -i 0x0000000c
```

Signed-off-by: deepakala-k <deepakala.karthikeyan@ibm.com>
Change-Id: I6d09ae39b19f0e4ea83d44a48083e959089818be

devenrao reviewed (commented):
Change your commit message as shown below ( you can change if required) but it should have more details so that the reviewer can easily understand the issue and the fix.
"
As part of the guard records list (guard -l) prefix 0x in front of the record-id so that the same can be used as part of "guard -i" command.

In the existing implementation guard -l will list the record id without 0x as a prefix example "abcd1234" and the user needs to prefix it with 0x manually to use with guard -i command
i.e guard -i 0xabcd1234

Tested
Existing implementation
root@rain104bmc:~# /tmp/guard -l
ID         | ERROR    | Type     | Path
10000000 | 00000000 | manual   | physical:sys-0/node-0/proc-1

After prefixing with 0x
root@rain104bmc:~# /tmp/guard -l
ID         | ERROR    | Type     | Path
0x10000000 | 00000000 | manual   | physical:sys-0/node-0/proc-1

user can now use 
guard -i 0x10000000

Fixed the formatting and spacing of the header of the guard -l command output.
"

RameshIyyar commented on guard.cpp:
```diff
@@ -11,19 +11,26 @@ using namespace openpower::guard;
 void printHeader()
 {
-    std::cout << "ID       | ERROR    |  Type  | Path " << std::endl;
+    std::cout << std::left << std::setfill(' ') << std::setw(10) << "ID";
```
Can you please push the header correction in the separate commit like @devenrao mentioned?

RameshIyyar commented on guard.cpp:
```diff
@@ -11,19 +11,26 @@ using namespace openpower::guard;
+    std::cout << " | ";
+    std::cout << std::left << std::setfill(' ') << std::setw(8) << "ERROR";
+    std::cout << " | ";
+    std::cout << std::left << std::setfill(' ') << std::setw(8) << "Type";
```
JS (Just Suggestion): FYI, right now the [guard reason string](https://github.com/open-power/guard/blob/main/libguard/guard_entity.hpp#L26) length is more than 8 characters, we can adjust it based on the max length of the reason but, nothing critical to change since it is just a cosmetic comment and `setw` won't skip value while printing the `Type` field value for each record if more than 8 characters.

For example:
```
ID       | ERROR    |  Type  | Path     <----------------------
00000001 | 89002d9e | predictive | physical:sys-0/node-0/proc-5/eq-5/fc-0/core-1
```

RameshIyyar commented on guard.cpp:
```diff
@@ -11,19 +11,26 @@ using namespace openpower::guard;
               << record.recordId;
 
     std::cout << " | ";
     std::cout << std::hex << std::setw(8) << std::setfill('0') << record.elogId;
```
I think we can add `0x` as a prefix for the elogid too since we are printing it as a hex value and its helps if we want to see the error log details through another tool (`peltool`)

RameshIyyar commented on guard.cpp:
```diff
@@ -11,19 +11,26 @@ using namespace openpower::guard;
 
```
**Commit message tips:**

- Try to follow the 50/72 rule for commit messages - that is, the subject line should not exceed 50 characters and the body should not exceed 72 characters. This is common practice in many projects which use Git.
```
guard -l command : Formatted the record ID to display 0x and the header space to align
```
- It will be great if you mentioned, "Why are you making changes, like issue details" instead of "How you changed" in the commit message.
- Please avoid adding the test machine name `root@rain104bmc` in the commit message.
- The `Change-Id:` is not required, it's used for the Gerrit review process, not for the pull request and, we are not using the Gerrit for the PHAL repository.

---------------------------------

Looks like the `ID` value is different when compared with the current code, `0x10000000` vs `00000001`, why is it like that?

deepakala-k replied:
This is addressed. It was due to adding std::left to header.

RameshIyyar commented on guard.cpp:
```diff
@@ -11,19 +11,26 @@ using namespace openpower::guard;
 void printRecord(const GuardRecord& record)
 {
-    std::cout << std::hex << std::setw(8) << std::setfill('0')
+    std::cout << "0x" << std::hex << std::setw(8) << std::setfill('0')
```
We could print it as a decimal value instead of a hex value but, it is ok.

deepakala-k committed:
guard: Align the guard list command header with its contents

- After adding 0x for RecordId and error logId the guard list command
header did not align with the contents

- So align the header and the information to help the user
understand clearly

Without fix
guard -l
ID       | ERROR    |  Type  | Path
0x00000001 | 0x00000000 | manual          | physical:sys-0/node-0/proc-1

After header alignment
guard -l
ID         | ERROR      | Type            | Path
0x00000001 | 0x00000000 | manual          | physical:sys-0/node-0/proc-1

Change-Id: I73895f200d4bb91eaf0b2a376c5d2321e59dd765
Signed-off-by: deepakala-k <deepakala.karthikeyan@ibm.com>
